driver/infoblox_ipam.go

```diff
@@ -76,10 +76,28 @@ func (ibDrv *InfobloxDriver) RequestAddress(r *ipamApi.RequestAddressRequest) (*
 	if rType, ok := r.Options["RequestAddressType"]; macAddr == "" || (ok && rType == netlabel.Gateway) {
 		logrus.Debugln("Request for Gateway IP")
 
+		l := &ibclient.NetworkViewLock{Name: network.NetviewName, ObjMgr: ibDrv.objMgr, LockEA: common.EA_DOCKER_PLUGIN_LOCK,
+			LockTimeoutEA: common.EA_DOCKER_PLUGIN_LOCK_TIME}
+		err := l.Lock()
+
+		if err != nil {
+			return  &ipamApi.RequestAddressResponse{}, fmt.Errorf("Failed to Allocate Gateway IP :%s", err)
+		}
+		defer l.UnLock(false)
+
+		if r.Address != "" {
+			ipAddr, err := ibDrv.objMgr.GetFixedAddress(network.NetviewName, network.Cidr, r.Address, "")
+			if err == nil && ipAddr != nil {
```

Comment:
This will only work for the networks which are created by the plugin. This will not work if there is already a existing network. For eg as in case of MACVLAN , if a user wants to use a existing network in docker, he/she will create network by passing the gateway ip which already exists in NIOS with some MAC address

```docker network create --config-only --subnet 172.28.128.0/24 --gateway 172.28.128.1 -o parent=eth1  mv-config```